### ISSUE TITLE
FW Upgrade issue

### ISSUE THREAD

**Issue by yursel** [Protocentral/protocentral_heartypatch#2]
I have tried to upgrade the firmware but no success.
Let me tell U the experience:

Linux machine, Ubuntu 16.04

-	xtensa-esp32-elf: I have tried both to build it (successfully) and downloading the pre-built version

-	esp-idf:  here is where the problem probably is.
On your instructions, you mention ( https://github.com/Protocentral/protocentral_heartypatch/blob/master/docs/firmware-upgrades.md ) to use “The v2.1 release of esp-idf”

If I use the checkout of v2.1, I get “make errors” in the app

If I use the checkout of V3.0-rc1, I can build, but I get an error while flashing.
I tried this use because in the dump of the device I see this is the version inside. Please give a look at the screenshot below.

If I use the checkout of V3.0 make app works, but it fails the bootloader make.

Basically, I have not been able to find a working configuration, nor to flash any firmware.

-------- ANSWER from Ashwin (very nice guy!)

I think the problem is with the toolchain. Espressif is not very clear about this. The latest version of the toolchain doesn't work well.

Can you please try out with the toolchain in the following link and let me know?

http://esp-idf.readthedocs.io/en/v2.1/get-started/linux-setup.html

The problem is that v2.1 seems to work only with this toolchain and the v3.0rc1 works with a specific version of the toolchain (not 0.80). 

-------- Update on my side

Proposed solution is working, I am able to compile.
